graylog2-web-interface/src/views/logic/views/normalizeViewState.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import keyBy from 'lodash/keyBy';
+
+import type ViewStateType from 'views/logic/views/ViewState';
+import { normalizeWidgetPositions } from 'views/logic/widgets/normalizeWidgetPositions';
+
+const normalizeViewState = (viewState: ViewStateType) => {
+  const { widgetPositions, widgets } = viewState;
+  const widgetById = keyBy(widgets.toArray(), 'id');
+  const normalizedWidgetPositions = normalizeWidgetPositions(widgetPositions, widgetById);
+
+  return viewState
+    .toBuilder()
+    .widgetPositions(normalizedWidgetPositions)
+    .build();
+};
+
+export default normalizeViewState;
```

Comment:
`normalizeViewState` sounds like we should call this function whenever we update the view state. Maybe name it more widget positions specific, like `normalizeWidgetPositionsInViewState`?

Reply:
I named it like that because `normalizeViewState` returns normalized `ViewState`. For positions, there is another function `normalizeWidgetPositions. So far we normalizing only the positions but in the future `normalizeViewState` might collect all other normalization functions.